File: programmers/python/level3/월간코드챌린지/풍선터트리기.py
## https://school.programmers.co.kr/learn/courses/30/lessons/68646
def solution(a):
    min_from_left = [0] * (len(a))
    min_from_right = [0] * (len(a))
    
    min_num = 1000000001
    for i in range(len(a)):
        if min_num > a[i]:
            min_num = a[i]
        min_from_left[i] = min_num
    
    min_num = 1000000001
    for i in range(len(a)-1, -1, -1):
        if min_num > a[i]:
            min_num = a[i]
        min_from_right[i] = min_num
    
    cnt = 0
    for i in range(len(a)):
        if min_from_left[i] == a[i] or min_from_right[i] == a[i]:
            cnt += 1
            
    return cnt

File: programmers/java/src/level3/월간코드챌린지/시즌1/풍선터트리기.java
package level3.월간코드챌린지.시즌1;

import java.util.Arrays;

// https://school.programmers.co.kr/learn/courses/30/lessons/68646
public class 풍선터트리기 {
    public int solution(int[] a) {
        int answer = 1;
        int M = Arrays.stream(a).min().getAsInt();
        int m = 1000000001;
        int i = 0;
        while(a[i] != M) {
            if(a[i] < m) {
                answer++;
                m = a[i];
            }
            i++;
        }
        m = 1000000001;
        i = a.length-1;
        while(a[i] != M) {
            if(a[i] < m) {
                answer++;
                m = a[i];
            }
            i--;
        }

        return answer;
    }
}
